apps/api/src/scraper/scrapeURL/engines/pdf/index.ts

```diff
@@ -316,125 +316,136 @@ export async function scrapePDF(meta: Meta): Promise<EngineScrapeResult> {
+    if (!result) {
+      result = await scrapePDFWithParsePDF(
         {
           ...meta,
           logger: meta.logger.child({
-            method: "scrapePDF/scrapePDFWithRunPodMU",
+            method: "scrapePDF/scrapePDFWithParsePDF",
           }),
         },
         tempFilePath,
-        base64Content,
-        maxPages,
-      );
-      const muV1DurationMs = Date.now() - muV1StartedAt;
-      meta.logger
-        .child({ method: "scrapePDF/MUv1Experiment" })
-        .info("MU v1 completed", {
-          durationMs: muV1DurationMs,
-          url: meta.rewrittenUrl ?? meta.url,
-          pages: effectivePageCount,
-          success: true,
-        });
-    } catch (error) {
-      if (
-        error instanceof RemoveFeatureError ||
-        error instanceof AbortManagerThrownError
-      ) {
-        throw error;
-      }
-      meta.logger.warn(
-        "RunPod MU failed to parse PDF (could be due to timeout) -- falling back to parse-pdf",
-        { error },
       );
-      Sentry.captureException(error);
-      const muV1DurationMs = Date.now() - muV1StartedAt;
-      meta.logger
-        .child({ method: "scrapePDF/MUv1Experiment" })
-        .info("MU v1 failed", {
-          durationMs: muV1DurationMs,
-          url: meta.rewrittenUrl ?? meta.url,
-          pages: effectivePageCount,
-          success: false,
-        });
     }
-  }
 
-  // If RunPod MU failed or wasn't attempted, use PdfParse
-  if (!result) {
-    result = await scrapePDFWithParsePDF(
-      {
-        ...meta,
-        logger: meta.logger.child({
-          method: "scrapePDF/scrapePDFWithParsePDF",
-        }),
+    return {
+      url: response.url ?? meta.rewrittenUrl ?? meta.url,
+      statusCode: response.status,
+      html: result?.html ?? "",
+      markdown: result?.markdown ?? "",
+      pdfMetadata: {
+        // Rust parser gets the metadata incorrectly, so we overwrite the page count here with the effective page count
+        // TODO: fix this later
+        numPages: effectivePageCount,
+        title: pdfMetadata.title,
       },
-      tempFilePath,
-    );
-  }
-
-  await unlink(tempFilePath);
 
-  return {
-    url: response.url ?? meta.rewrittenUrl ?? meta.url,
-    statusCode: response.status,
-    html: result?.html ?? "",
-    markdown: result?.markdown ?? "",
-    pdfMetadata: {
-      // Rust parser gets the metadata incorrectly, so we overwrite the page count here with the effective page count
-      // TODO: fix this later
-      numPages: effectivePageCount,
-      title: pdfMetadata.title,
-    },
-
-    proxyUsed: "basic",
-  };
+      proxyUsed: "basic",
+    };
+  } finally {
+    // Always clean up temp file after we're done with it
+    try {
+      await unlink(tempFilePath);
+    } catch (error) {
+      // Ignore errors when cleaning up temp files
+      meta.logger?.warn("Failed to clean up temporary PDF file", {
+        error,
+        tempFilePath,
+      });
+    }
```

Comment:
diff looks weird, but basically added this try block with this finally statement to delete files